src/Service/ResourceFieldPopulator.php

```diff
@@ -0,0 +1,151 @@
+     * Throws an exception when the CKAN API is unreachable or responds with an error or if the resource is not properly
+     *
+     * @param Resource $resource
+     * @throws \GuzzleHttp\Exception\GuzzleException|RuntimeException
+     */
+    public function populateFields(Resource $resource)
+    {
+        if (!$resource->Endpoint && !$resource->Identifier) {
+            throw new RuntimeException('Could not fetch fields for a resource that is not fully configured');
+        }
+
+        $fieldSpecs = $this->doFieldRequest($resource);
+
+        $newFields = [];
+        $fields = $resource->Fields();
+
+        foreach ($fieldSpecs as $fieldSpec) {
+            $id = $fieldSpec['id'];
+
+            // Skip fields that may already exist
+            if ($fields->find('Name', $id)) {
+                continue;
+            }
+
+            // Create a new field
+            $newFields[] = $field = ResourceField::create();
+            $field->Name = $id;
+            // Attempt to parse a readable name
+            $field->ReadableName = $this->parseName($id);
+            $field->Type = $fieldSpec['type'];
+        }
+
+        // Add our new fields
+        $fields->addMany($newFields);
+    }
+
+    /**
+     * Perform a request to the CKAN endpoint provided by the given resource to fetch field definitons
+     *
+     * @param Resource $resource
+     * @return mixed
+     * @throws \GuzzleHttp\Exception\GuzzleException|RuntimeException
+     */
+    protected function doFieldRequest(Resource $resource)
+    {
+        $endpoint = sprintf(
+            '%s/api/3/action/datastore_search?id=%s',
+            trim($resource->Endpoint, '/'),
+            $resource->Identifier
+        );
+
+        $request = new Request('GET', $endpoint);
+        $response = $this->getGuzzleClient()->send($request, $this->getClientOptions());
+
+        $statusCode = $response->getStatusCode();
+        if ($statusCode < 200 || $statusCode >= 300) {
+            throw new RuntimeException('CKAN API is not available. Error code ' . $statusCode);
+        }
+
+        if (!count(preg_grep('#application/json#', $response->getHeader('Content-Type')))) {
+            throw new RuntimeException('CKAN API returns an invalid response: Content-Type is not JSON');
+        }
+
+        $responseBody = json_decode($response->getBody()->getContents(), true);
+
+        if (!$responseBody || !isset($responseBody['success']) || !$responseBody['success']) {
+            throw new RuntimeException('CKAN API returns an invalid response: Responded as invalid');
+        }
+
+        return $responseBody['result']['fields'];
+    }
+
+    /**
+     * Parse given column ID for a more readable version
+     *
+     * @param $id
+     * @return string
+     */
+    protected function parseName($id)
+    {
+        // Parse "camelCase" to "space case"
+        $name = strtolower(preg_replace('/(?<=[a-z\d])([A-Z])/', ' \1', $id));
```

Comment:
We have `ResourceField::generateReadableName()` doing this logic as of #40, this looks better though. Can you consolidate them?